netifd: fix typo (timesvr) in dhcp script

senlisu opened the pull request:
Match DHCP Option 4 handling  to udhcpc's environment variable (timesrv).

Signed-off-by: Sukru Senli <sukru.senli@iopsys.eu>

senlisu committed:
netifd: fix typo (timesvr) in dhcp script

Correct /lib/netifd/dhcp.script DHCP Option 4 (Time Server) handling by matching the variable name to udhcpc's environment variable (timesrv).

adrianschmutzler commented:
Your commit has formal issues. Please properly wrap the commit message after 74 chars and add a Signed-off-by line.

adrianschmutzler commented:
Since people are using this, the question is
- whether this should be fixed at all
- whether we provide migration if fixed
- whether this should be backported

(I just think of HTTP_REFERER here ...)

ynezz commented:
What about supporting both options for some time, filling syslog with some deprecation/warning message for the variable name with typo?

senlisu commented:
> Your commit has formal issues. Please properly wrap the commit message after 74 chars and add a Signed-off-by line.

Thanks. The requested change to commit message is done.

adrianschmutzler commented:
> > Your commit has formal issues. Please properly wrap the commit message after 74 chars and add a Signed-off-by line.
> 
> Thanks. The requested change to commit message is done.

No, they aren't? Be aware that I'm talking about the commit, not the PR comments.

adrianschmutzler commented:
> What about supporting both options for some time, filling syslog with some deprecation/warning message for the variable name with typo?

That would be the best way to do it IMO, but it will mean additional work for the submitter.

senlisu committed:
netifd: indicate upcoming deprecation of 'timesvr'

/lib/netifd/dhcp.script:
    Keep support for 'timesvr' while also supporting 'timesrv'
    Add log message indicating deprecation of 'timesvr'

Signed-off-by: Sukru Senli <sukru.senli@iopsys.eu>

senlisu commented:
> > What about supporting both options for some time, filling syslog with some deprecation/warning message for the variable name with typo?
> 
> That would be the best way to do it IMO, but it will mean additional work for the submitter.

Would the second commit be sufficient for the fix?

senlisu commented:
Sorry guys, by mistake I deleted my branch which closed this issue. So, I had to create a new pull request, hopefully this time with correct details.
